lib/middleware/fireAlarmMiddleware.ts

```diff
@@ -0,0 +1,193 @@
+import { Arguments, terminalWidth } from 'yargs';
+import chalk from 'chalk';
+import { fetchFireAlarms } from '@hubspot/local-dev-lib/api/fireAlarm';
+import { FireAlarm } from '@hubspot/local-dev-lib/types/FireAlarm';
+import { logger } from '@hubspot/local-dev-lib/logger';
+import { debugError } from '../errorHandlers';
+import pkg from '../../package.json';
+import { UI_COLORS } from '../ui';
+import { i18n } from '../lang';
+
+/*
+ * Versions can be formatted like this:
+ * =7.2.2 -> targets the exact version 7.2.2
+ * =7.2.* -> targets all versions with a major of 7 and a minor of 2
+ * =7.* -> targets all versions with a major of 7
+ * =* -> targets all versions
+ * <=7.2.2 -> targets all versions equal to or less than 7.2.2
+ * <=7.2.* -> targets all versions equal to or less than 7.2
+ */
+const WILDCARD = '*';
+
+function isVersionTargeted(
+  version: string,
+  targetVersionString: string | null
+): boolean {
+  // Assume we're targeting no versions if no version string is found
+  // Target all versions using the * wildcard
+  if (!targetVersionString) {
+    return false;
+  }
+
+  // Ignore alerts on any version tags (like -beta)
+  // No need for us to send alerts for these beta CLI versions
+  if (version.includes('-') || targetVersionString.includes('-')) {
```

Comment:
Hmm do you think we'll never want to target beta versions? I could see it being helpful occasionally. Maybe we could exclude them from the more generalized targeting (with `*` and `>=` etc) but let you target them by specifying the exact version?

Reply:
I like that idea. This was somewhat of an arbitrary call on my part. I'll add in some support for exact version targeting for beta versions.